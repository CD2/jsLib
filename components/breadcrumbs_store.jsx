import React from 'react'
import { Link } from 'react-router-dom';
import { history } from 'utils/router'
import home from 'images/home.svg'
import { Grid, GridItem } from 'lib/components/grid'
import { styled, t } from 'utils/theme'
import { small_icon } from 'utils/common_styles'
import Wrapper from "./wrapper";
import Img from "./img";

import BreadcrumbStore from 'stores/breadcrumbs'
import { observer } from 'mobx-react'
import decorate from 'utils/decorate'

export class Breadcrumbs extends React.Component {

  static defaultProps = {
    breadcrumbs: [],
  }

  renderBreadcrumb({name, href}) {
    return (
      <span key={href}>
        <span> > </span>
        <Link className='breadcrumb__link' to={href}>{name}</Link>
      </span>
    )
  }

  render() {
    if (BreadcrumbStore.breadcrumbs.filter(crumb=>crumb).length === 0) return null
    return(
      <Wrapper width="1600" className={this.props.className} background={this.props.theme.secondary}>
        <Grid>
          <GridItem weight={5/6}>
            <span>
              <Link className='breadcrumb__link' to='/'>
                <Img src={home} />
              </Link>
            </span>
            {BreadcrumbStore.breadcrumbs.filter(crumb=>crumb).map(this.renderBreadcrumb)}
          </GridItem>
          <GridItem align="right" weight={1/6}>
            <a onClick={ history.goBack } >Back</a>
          </GridItem>
        </Grid>
      </Wrapper>
    )
  }
}
export default decorate(
  styled`
    color: ${t('lightText')};
    box-shadow: ${t('shadow1')};
    color: white;
    font-weight: 600;
    .breadcrumb__link {
      border-bottom: 0;
    }
    img {
      ${small_icon}
    }
    a {
      color: white;
    }
  `,
  observer,
  Breadcrumbs
)
