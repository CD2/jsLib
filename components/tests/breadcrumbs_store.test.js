import t from 'utils/test_helper'
import BreadcrumbStore from 'stores/breadcrumbs'
import { Breadcrumbs } from '../breadcrumbs_store'

t.describe_component(Breadcrumbs, { theme: { secondary: 1 }}, (c) => {
  c.call_function(`renderBreadcrumb`, { name: ``, href: `` })
  BreadcrumbStore.set(1,{ name: `b`, href: `/1` })
  c.has_css(`Wrapper`).has_css(`Grid`, (c) => {
    c.has_css(`GridItem`).with_attr(`weight`, 5/6).has_css(`span`, (c) => {
      c.has_css(`a`).with_attr(`className`, `breadcrumb__link`)
    })
    c.has_css(`GridItem`).with_attr(`align`, `right`).with_attr(`weight`, 1/6)
  })
})
