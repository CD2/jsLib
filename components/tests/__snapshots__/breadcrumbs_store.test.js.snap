// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component <Breadcrumbs /> renders correctly 1`] = `
<withTheme(StyledComponent)
  background={1}
  width={1600}
>
  <Grid>
    <GridItem
      weight={0.8333333333333334}
    >
      <span>
        <Link
          className="breadcrumb__link"
          replace={false}
          to="/"
        >
          <withTheme(StyledComponent)
            icon="home"
          />
        </Link>
      </span>
      <span>
        <span>
           &gt; 
        </span>
        <Link
          className="breadcrumb__link"
          replace={false}
          to="/1"
        >
          b
        </Link>
      </span>
    </GridItem>
    <GridItem
      align="right"
      weight={0.16666666666666666}
    >
      <a
        onClick={[Function]}
      >
        Back
      </a>
    </GridItem>
  </Grid>
</withTheme(StyledComponent)>
`;
